bastion/src/macros.rs

```diff
@@ -0,0 +1,241 @@
+                    async move {
+                        let ctx = ctx;
+                        loop {
+                            let msg = ctx.recv().await?;
+                            ($action)(msg);
+                        }
+                    }
+                })
+        }).expect("failed to create children group");
+    };
+}
+
+/// This macro creates a new supervisor with the given strategy and the given callbacks.
+/// Children can be specified by using the `children` / `child` macro.
+/// You can provide as many children groups as you want. Supervised supervisors are currently not
+/// yet supported.
+///
+/// # Example
+/// ```
+/// # use bastion::prelude::*;
+/// # fn main() {
+/// let sp = supervisor! {
+///     callbacks: Callbacks::default(),
+///     strategy: SupervisionStrategy::OneForAll,
+/// };
+/// # }
+/// ```
+#[macro_export]
+macro_rules! supervisor {
+    ($($keys:ident: $vals:expr,)*) => {
+        supervisor!(@sort,
+            $crate::supervisor::SupervisionStrategy::OneForAll,
+            $crate::Callbacks::default(),
+            $($keys: $vals,)*
+        )
+    };
+
+    (@sort,
+    $strat:expr, $_:expr,
+    callbacks: $cbs:expr,
+    $($keys:ident: $vals:expr,)*) => {
+        supervisor!(@sort,
+            $strat,
+            $cbs,
+            $($keys: $vals,)*
+        )
+    };
+
+    (@sort,
+    $_:expr, $cbs:expr,
+    strategy: $strat:expr,
+    $($keys:ident: $vals:expr,)*) => {
+        supervisor!(@sort,
+            $strat,
+            $cbs,
+            $($keys: $vals,)*
+        )
+    };
+
+    (@sort, $strat:expr, $cbs:expr,) => {
+        $crate::Bastion::supervisor(|sp| {
+            sp
+                .with_callbacks($cbs)
+                .with_strategy($strat)
+        }).expect("failed to create supervisor");
+    };
+}
+
+/// Spawns a blocking task, which will run on the blocking thread pool,
+/// and returns the handle.
+///
+/// # Example
+/// ```
+/// # use std::{thread, time};
+/// # use lightproc::proc_stack::ProcStack;
+/// # use bastion::prelude::*;
+/// # fn main() {
+/// let task = blocking! {
+///     thread::sleep(time::Duration::from_millis(3000));
+/// };
+/// run!(task);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! blocking {
+    ($($tokens:tt)*) => {
+        bastion_executor::blocking::spawn_blocking(async move {
+            $($tokens)*
+        }, lightproc::proc_stack::ProcStack::default())
+    };
+}
+
+/// This macro blocks the current thread until passed
+/// future is resolved with an output (including the panic).
+///
+/// # Example
+/// ```
+/// # use bastion::prelude::*;
+/// # fn main() {
+/// let future1 = async move {
+///     123
+/// };
+///
+/// run! {
+///     let result = future1.await;
+///     assert_eq!(result, 123);
+/// };
+///
+/// let future2 = async move {
+///     10 / 2
+/// };
+///
+/// let result = run!(future2);
+/// assert_eq!(result, 5);
+/// # }
+/// ```
+#[macro_export]
+macro_rules! run {
```

Comment:
Could you also add a `spawn!` macro that would call `bastion_executor::pool::spawn`, please? :)